packages/@aws-cdk/cdk/lib/intrinsic.ts

```diff
@@ -0,0 +1,33 @@
+import { IResolveContext, Token } from "./token";
+
+export interface IntrinsicProps {
+  /**
+   * A hint for the Token's purpose when stringifying it
+   */
+  readonly displayHint?: string;
+}
+
+/**
+ * Token subclass that represents values intrinsic to the target document language
+ */
+export class Intrinsic extends Token {
```

Comment:
TBH I am not sure why this is needed? Kind of feels like `encodeAsXxx` should accept `any` which can be either an `IToken` or some primitive values that will be passed through as a token.

I also don't like the CFN semantics that "seeped through" to this layer. If you insist on keeping this class (again, I don't think it's really needed), then I would call it `AnyValue` or `AnyToken` or something like that.

Reply:
The word  "intrinsic" happens to exist in CloudFormation already, but I actually think it makes sense to have this as a concept in CDK: a value that makes sense in the target  document language, which might not strictly adhere to the typing of the host language. The  document language might be CFN, might be Google Cloud Manager,  might be Kubernetes YAML, might be anything. There will still be value in the "intrinsic" concept, and nobody is forcing us to call it something else than it is called  in CloudFormation. 

As a concept name, I like "intrinsic" better than "AnyValue". 

I also don't see why we need to bind the representation of the "AnyValue" and the encoding together. To me, it feels better to make them explicitly two operations: first represent a document language-level value, and *maybe* additionally encode it.

Users won't need to see any of this anyway, they will have  the necessary operations pre-wrapped, and only use the `Lazy` constructors.